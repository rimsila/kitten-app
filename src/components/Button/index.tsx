import { CusBox, CusText } from 'components';
import React, { ReactNode } from 'react';
import { TouchableOpacity } from 'react-native';

export type ButtonProps = React.ComponentProps<typeof CusBox> & {
  onPress?: () => void;
  label?: string;
  children?: ReactNode;
  labelProps?: React.ComponentProps<typeof CusText>;
};

const Button = ({ onPress, label, children, labelProps, ...rest }: ButtonProps) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <CusBox {...rest}>
        {typeof label === 'string' ? (
          <CusText variant="subheader" {...labelProps}>
            {label}
          </CusText>
        ) : (
          children
        )}
      </CusBox>
    </TouchableOpacity>
  );
};

export default Button;
