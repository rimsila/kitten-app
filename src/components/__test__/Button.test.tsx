import { render } from '@testing-library/react-native';
import Button from 'components/Button';
import React from 'react';

describe('<Button />', () => {
  it('should match snapshot', () => {
    const rendered = render(<Button />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
