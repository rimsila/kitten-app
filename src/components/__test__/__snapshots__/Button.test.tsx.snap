// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
      ]
    }
  />
</View>
`;
